### PR TITLE
Dev/qwen omni

### PR DESCRIPTION
**Pull request by ngquangtrung57**
## Motivation
Support Qwen 2.5 Omni

## Modifications
- Add Qwen 2.5 Omni 
- Add video backend: qwen_omni_utils

## Checklist

- [ ] Format your code
- [ ] Add unit tests
- [ ] Update documentation as needed, including docstrings or example tutorials.

### CODE REVIEW

**Review by kcz358 — COMMENTED** [review]
Hi, I think most of the parts look fine right now. A few suggestion that can improve this PR further. I see that we are only training the Thinker part so I think you can wrap the actual training model into a PretrainedModel and save if as a pretrained checkpoint. Then you can simply use `from_pretrained` using that and rename the model into `qwen2_5_omni_thinker` (can push the converted checkpoint to hub also, in lmms-lab maybe)

**Comment by kcz358** [comment]
To clarify more, my feeling is that you can actually dump away all the talker module. You can do it as follow

You can see here that thinker module itself is a pretrainedmodel and has its own model type
https://github.com/huggingface/transformers/blob/94df0e65602922be2831b3faa457a2bde78b936b/src/transformers/models/qwen2_5_omni/configuration_qwen2_5_omni.py#L413-L484

So you can
1. First load pretrained from original Qwen2.5-Omni checkpoints
2. Extract the thinker module (Can be like model.thinker, model.thinker.thinker_config)
3. Save the pretrained thinker checkpoints, config, and preprocessor using `save_pretrained`
4. In the init part, register the thinker module as an `AutoModelForCausalLM` (it is not registered by the transformers right?)
5. Load from your pretrained checkpoints, processor, and config so that we have a very clean mapping func and register
6. [Optional] Push the checkpoints to the hub so that everyone can use it

You might need to change your patching logic accordingly though, because the model structure changes a bit

**Comment by kcz358** [comment]
> * Most rmpad logic falls under liger file not the ops file because Qwen Omni doesn't have intermediate model layer.
> * I have some issues with the register and loading of Thinker only, so i still modify the mapping_func file.
> * I have fixed the issue with tmrope that caused error with sp i mentioned, but currently with my testing using sp=2 on 8 GPUs, it ran successfully the first step. Then at the second step, it got nccl timeout due to AlltoAll operation. This could be my server issue because i have this same error for different projects, when the GPUs perform AllReduce or Allgather.

For the sp alltoall ops, this could possibly caused by the tensor shape mismatch during the gather operations so it is stucked there. May need to check the splitted size on each rank and validate whether the gather size is being set correctly

**Comment by ngquangtrung57** [comment]
I have extracted the thinker only and pushed it here "ngqtrung/Qwen2.5-Omni-Thinker-7B". When i move it to lmms lab hub, it show "The storage patterns of org lmms-lab tripped our internal systems! Please contact us at website@huggingface.co so we can verify your account and unlock more storage for your use-case." so currently, i keep it in my hub. 
I have checked the tensor shape of AlltoAll op and it looks correct. also the hang happens after lce forward(after compte the loss) not when performing "gather_heads_scatter_seq". i think this could be related to trainer or as i said, my server itself.

**Review by kcz358 — APPROVED** [review]

**Comment by kcz358** [comment]
Oh, actually on my testing envs, this also breaks some of my cicd in some cases. Let me see if we can fix it together in a new PR
